lib/api/core/plugins/pluginsManager.js

```diff
@@ -473,295 +636,269 @@ function initControllers (controllers, routes, plugin, pluginName) {
-    }
+      if (!isObject(strategy.config.authenticateOptions)) {
+        // eslint-disable-next-line no-console
+        console.warn(clc.yellow(`${errorPrefix} The "${clc.bold('authenticateOptions')}" of the strategy "${clc.bold(strategyName)}" must be an object.`));
+        valid = false;
+      }
 
-    try {
-      const loader = packageExists ? loadPluginFromPackageJson : loadPluginFromDirectory;
+      if (!strategy.config.fields || !Array.isArray(strategy.config.fields)) {
+        // eslint-disable-next-line no-console
+        console.warn(clc.yellow(`${errorPrefix} The "${clc.bold('fields')}" of the strategy "${clc.bold(strategyName)}" must be an array.`));
+        valid = false;
+      }
 
-      plugin = loader(path.resolve(pluginsDir, pluginDirName), config);
-    } catch (e) {
-      console.error(`Unable to load plugin ${pluginDirName}. ${e.message}\n${e.stack}`); // eslint-disable-line no-console
-      return;
     }
-
-    loadedPlugins[plugin.name] = plugin;
   });
 
-  return loadedPlugins;
-}
-
-function loadPluginFromPackageJson(pluginPath, config) {
-  const packageJson = require(path.resolve(pluginPath, 'package.json'));
-  const PluginClass = require(pluginPath);
+  if (valid) {
+    Object.keys(plugin.object.strategies).forEach(strategyName => {
+      const
+        strategy = plugin.object.strategies[strategyName],
+        methods = {};
 
-  debug('[%s] loading plugin from package.json file in directory "%s"', packageJson.name, pluginPath);
+      Object.keys(strategy.methods).filter(name => name !== 'verify').forEach(methodName => {
+        methods[methodName] = (...args) => {
+          return new Bluebird((resolve, reject) => {
+            try {
+              resolve(plugin.object[strategy.methods[methodName]].bind(plugin.object)(...args));
+            }
+            catch (error) {
+              reject(error instanceof KuzzleError ? error : new PluginImplementationError(error));
+            }
+          });
+        };
+      });
 
-  return {
-    name: packageJson.name,
-    object: new PluginClass(),
-    config: config[packageJson.name] || {},
-    path: pluginPath
-  };
-}
+      pluginsManager.authentications[strategyName] = {
+        strategy,
+        methods
+      };
 
-function loadPluginFromDirectory(pluginPath, config) {
-  const PluginClass = require(pluginPath);
-  const pluginName = path.basename(pluginPath);
+      try {
+        registerStrategy(pluginsManager, plugin, strategyName);
+        pluginsManager.kuzzle.passport.injectAuthenticateOptions(strategyName, strategy.config.authenticateOptions || {});
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error(e.message);
 
-  debug('[%s] loading plugin from directory "%s"', pluginName, pluginPath);
+        valid = false;
+      }
+    });
+  }
 
-  return {
-    name: pluginName,
-    object: new PluginClass(),
-    config: config[pluginName] || {},
-    path: pluginPath
-  };
+  return valid;
```

Comment:
no need to return `valid` variable: if the method did not throw any exception, it means that the authentication strategies have been correctly initialized